plugin.video.lbry/resources/lib/local.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+import xbmc
+import xbmcaddon
+import xbmcvfs
+import xbmcgui
+
+ADDON = xbmcaddon.Addon()
+tr = ADDON.getLocalizedString
+
+def get_profile_path(rpath):
+    return xbmc.translatePath(ADDON.getAddonInfo('profile') + rpath)
+
+def load_channel_subs():
+    channels = []
+    try:
+        f = xbmcvfs.File(get_profile_path('channel_subs'), 'r')
+        lines = f.readBytes()
+        f.close()
+    except Exception as e:
+        pass
+    lines = lines.decode('utf-8')
+    for line in lines.split('\n'):
+        items = line.split('#')
+        if len(items) < 2:
+            continue
+        channels.append((items[0],items[1]))
+    return channels
+
+def save_channel_subs(channels):
+    try:
+        f = xbmcvfs.File(get_profile_path('channel_subs'), 'w')
```

Comment:
jfyi `xbmcvfs.File` does support context lib, so you should be able to use:

```python
with xbmcvfs.File(get_profile_path('channel_subs'), 'w') as f:
    f.something()
``` 

Not mandatory to change though.